libfwupd/fwupd-client.c

```diff
@@ -1007,6 +1007,8 @@ fwupd_client_disconnect(FwupdClient *self, GError **error)
 static void
 fwupd_client_quit_cb(GObject *source, GAsyncResult *res, gpointer user_data)
 {
+	FwupdClient *self = FWUPD_CLIENT(g_task_get_source_object(G_TASK(user_data)));
```

Comment:
This isn't a memory leak now?  I was worried that the g_autoptr(GTask) is using the same reference.

Reply:
nah, i think it's fine -- just two things pointing to the same address